coprocessor/fhevm-engine/fhevm-listener/tests/integration_test.rs

```diff
@@ -161,13 +161,21 @@ async fn test_listener_restart() -> Result<(), anyhow::Error> {
         end_at_block: None,
         catchup_margin: 5,
         log_level: Level::INFO,
-        health_port: 8080,
+        health_port: 8082,
     };
 
     // Start listener in background task
     let listener_handle = tokio::spawn(main(args.clone()));
 
-    tokio::time::sleep(tokio::time::Duration::from_secs(5)).await;
+    for delay in 0..60 {
```

Comment:
Is it doable to make this logic commonso we can automate the health-checking unit-tests in the other workers as well?

Reply:
yes I can add it in fhevm-engine-common.

Reply:
Done

Reply:
I ve done it but it added a big dependency to common. So I moved the functions back to the integration tests in fhevm-listener.
I am not sure how to have the function in common and only add the dependency in dev profile in common.